.github/workflows/release.yml

```diff
@@ -56,10 +58,31 @@ jobs:
           tag-semver: |
             {{version}}
       
-      - name: Print Tag info
+      - name: Print Build info
         run: |
           echo "${{ steps.docker_meta.outputs.tags }}"
           echo "RELEASE TAG: ${RELEASE_TAG}"
+          echo "RELEASE BRANCH: ${BRANCH}"
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Update Operator YAML
+        run: |
+          echo "update the operator yaml with the image tag"
+          sed -i "s/zfs-driver:ci/zfs-driver:${RELEASE_TAG}/" deploy/yamls/zfs-driver.yaml
+          sed -i "s/zfs-driver:ci/zfs-driver:${RELEASE_TAG}/" deploy/zfs-operator.yaml
+          sed -i "s/newTag: ci/newTag: ${RELEASE_TAG}/" deploy/yamls/kustomization.yaml
+
+          git diff --quiet deploy/ || CHANGED=$?
+          if [[ $CHANGED -ne 0 ]]; then
+            echo "Commiting operator yaml changes"
+            git add deploy/
+            git commit -s -m 'chore(yaml): updating operator yaml with the image tag'
+            git push origin HEAD:"${BRANCH}"
```

Comment:
@pawanpraka1 Was this tested?

The HEAD will be in a detached state, so how will this push work?

Reply:
yes, I have tested it, it works. It will push to the target branch mentioned in the command.